gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -150,7 +152,8 @@ fn sha256(data: &[u8]) -> sha256::Hash {
 fn get_payment_data(gateway: Option<LightningGateway>, invoice: Bolt11Invoice) -> PaymentData {
     match gateway {
         Some(g) if g.supports_private_payments => {
-            let pruned_invoice: PrunedInvoice = invoice.try_into().expect("Invoice has amount");
+            let pruned_invoice =
+                PrunedInvoice::try_from(&invoice, None).expect("Invoice has amount");
```

Comment:
I'd avoid calling the function the same as a widely-used trait. Maybe `(try_)from_invoice`?